src/main/java/ch/njol/skript/expressions/ExprQueue.java

```diff
@@ -0,0 +1,95 @@
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.registrations.Feature;
+import ch.njol.skript.util.LiteralUtils;
+import ch.njol.util.Kleenean;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+import org.skriptlang.skript.lang.util.SkriptQueue;
+
+import java.util.Iterator;
+
+@Name("Queue (Experimental)")
+@Description("""
+	Requires the <code>using queues</code> experimental feature flag to be enabled.
+	
+	Creates a new queue.
+	A queue is a set of elements that can have things removed from the start and added to the end.
+	
+	Any value can be added to a queue. Adding a non-existent value (e.g. `{variable that isn't set}`) will have no effect.
+	This means that removing an element from the queue will always return a value <i>unless the queue is empty</i>.
+	
+	Requesting an element from a queue (e.g. `the 1st element of {queue}`) also removes it from the queue.""")
+@Examples({
+	"""
+		set {queue} to a new queue
+		add "hello" and "there" to {queue}
+		broadcast the first element of {queue} # hello
+		broadcast the first element of {queue} # there
+		# queue is now empty""",
+	"""
+		set {queue} to a new queue of "hello" and "there"
+		broadcast the last element of {queue} # removes 'there'
+		add "world" to queue
+		broadcast the first 2 elements of {queue} # removes 'hello', 'world'"""
+})
+@Since("INSERT VERSION (experimental)")
+public class ExprQueue extends SimpleExpression<SkriptQueue> {
+
+	static {
+		Skript.registerExpression(ExprQueue.class, SkriptQueue.class, ExpressionType.SIMPLE,
+			"[a] new queue [contents:of %-objects%]");
```

Comment:
```suggestion
			"[a] new queue [contents:(of|with) %-objects%]");
```
What do you think of `a new queue with 1, 2, and 3`? I think some options for `of` would be good.

I'm also a bit unsure about requiring (and even having) `new` as a keyword in the syntax.

Reply:
We can't have `queue` on its own because that's ambiguous with the type name, so either you'd need to mandate having some initial items (so you can't create an empty queue and then add stuff to it) or you need an extra word in there somewhere. I went for the word.